streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/python/PythonGateway.java

```diff
@@ -0,0 +1,151 @@
+package org.ray.streaming.runtime.python;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.msgpack.core.Preconditions;
+import org.ray.api.annotation.RayRemote;
+import org.ray.streaming.api.context.StreamingContext;
+import org.ray.streaming.python.PythonFunction;
+import org.ray.streaming.python.PythonPartition;
+import org.ray.streaming.python.stream.PythonStreamSource;
+import org.ray.streaming.runtime.util.ReflectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Gateway for streaming python api.
+ * All calls on DataStream in python will be mapped to DataStream call in java by this
+ * PythonGateway using ray calls.
+ * <p>
+ * Note: this class needs to be in sync with `ray.streaming.runtime.gateway_client.GatewayClient`
```

Comment:
```suggestion
 * Note: this class needs to be in sync with `GatewayClient` in `/path/to/file.py`.
```
This would be easier for readers to find the corresponding Python class.

Reply:
fixed